### DISCUSSION TITLE
Style or ContolTheme, which one to use?

### DISCUSSION THREAD

**Discussion by ds1709** [AvaloniaUI/Avalonia#17629] [Q&A] [answered]
I have an application, where I want to override default style of some control, e.g., I want all my menu items to have less vertical padding.
So, there's to ways to do this:
```xaml
<Application>
    <Application.Resources>
        <ControlTheme x:Key="{x:Type MenuItem}" TargetType="{x:Type MenuItem}" BasedOn="{StaticResource {x:Type MenuItem}}">
            <Setter Property="Padding" Value="11 4" />
        </ControlTheme>
    </Application.Resources>
</Application>
```
or
```xaml
<Application>
    <Application.Styles>
        <Style Selector="MenuItem">
            <Setter Property="Padding" Value="11 4" />
        </Style>
    </Application.Styles>
</Application>
```
Which one is the most prefered?

**Comment by LaurentInSeattle** [comment]
Hi, 
If you have some WPF experience, ControlTheme should be your preference.
If you are more into CSS, use Styles. 

As far as I can tell, you can do everything using both methods.
Your code, your choice... :)

**Reply by timunie** [reply]
Main difference is that style cannot be overriden completely as it has no BasedOn option, unlike ControlTheme. I prefer Styles for tiny adjustments and ControlTheme when I want to style a lot or doing it in a Controls Library.

**Comment by maxkatz6** [comment]
If you task is to create a custom control look, replacing existing one - use ControlTheme.
If your task is to adjust styling a little bit with setters - use Style.

**Comment by ds1709** [comment, marked as answer]
I discovered (for myself) one important point to chose between `ControlTheme` and `Style`. If you want to customize some contol for entire application, it's better to use `ControlTheme` rather than `Style`. The reason is that `Style` always overrides `ControlTheme` values. It may become a problem if you are customizing `ItemsControl`s containers, like `ListBoxItem` and so on.

For example, if you have global (application level) `Style` for `ListBoxItem`
```xaml
<Style Selector="ListBoxItem">
    <Setter Property="Padding" Value="20 5" />
</Style>
```
and then somwhere in application create a `ListBox` with custom `ItemContainetTheme` like this
```xaml
<ListBox>
    <ListBox.ItemContainerTheme>
        <ControlTheme TargetType="{x:Type ListBoxItem}" BasedOn={StaticResource {x:Type ListBoxItem}}>
            <Setter Property="Padding" Value="0" />
        <ControlTheme />
    </ListBox.ItemContainerTheme>
</ListBox>
```
the `Padding` property will be `20,5` as defined in `Style`, and you have to define `Style` to override it.
But if you override base theme for `ListBoxItem` on application level (like this)
```xaml
<ControlTheme x:Key="{x:Type ListBoxItem}" TargetType="{x:Type ListBoxItem}" BasedOn={StaticResource {x:Type ListBoxItem}}>
    <Setter Property="Padding" Value="20 5" />
</ControlTheme>
```
all works as expected.
So, for me from now the main point to keep in mind to chose between `Style` and `ControlTheme` is theirs priority.
